### PR TITLE
Add IndexVersion to node info

### PR DESCRIPTION
**Pull request by thecoop**
Add an `index_version` field to node info to expose the current indexversion

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java [hunk 1]
```diff
@@ -64,6 +66,11 @@ public NodeInfo(StreamInput in) throws IOException {
         } else {
             transportVersion = TransportVersion.fromId(version.id);
         }
+        if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_500_075)) {
+            indexVersion = IndexVersion.readVersion(in);
+        } else {
+            indexVersion = IndexVersion.fromId(version.id);
```
Comment (thecoop) [inline server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/NodeInfo.java:72]: This is ok without any further checks. All released versions <8.11 have version-indexversion bijections, and serverless will quickly be updated to versions that know about this. The worst case is that the reported index version is incorrect, but this value isn't actually used anywhere.

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/docs

**Comment by thecoop** [comment]
@elasticmachine update branch

File: server/src/main/java/org/elasticsearch/TransportVersions.java [hunk 2]
```diff
@@ -138,6 +138,7 @@ static TransportVersion def(int id, String uniqueId) {
     public static final TransportVersion V_8_500_072 = def(8_500_072, "e2df7d80-7b74-4afd-9734-aee0fc256025");
     public static final TransportVersion V_8_500_073 = def(8_500_073, "9128e16a-e4f7-41c4-b04f-842955bfc1b4");
     public static final TransportVersion V_8_500_074 = def(8_500_074, "aab0c31c-62d7-4b95-bb29-0b6f367ece64");
+    public static final TransportVersion V_8_500_075 = def(8_500_075, "bd553a42-d9be-497f-aab2-b6be9e26bb4e");
```
Comment (ldematte) [inline server/src/main/java/org/elasticsearch/TransportVersions.java:141]: Suggestion: given https://github.com/elastic/elasticsearch/pull/99500 is about to be merged, maybe you can "plan ahead" and give this version a meaningful name?

**Review by ldematte — APPROVED** [review]
LGTM, with one minor suggestion

File: server/src/main/java/org/elasticsearch/TransportVersions.java (hunk 2 above)
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/TransportVersions.java:141]: I would prefer to keep these two separated - if this is merged first, #99500 should be updated to rename it. If #99500 is merged first, I can update this branch with the new scheme.

**Comment by thecoop** [comment]
@elasticmachine update branch

**Review by rjernst — APPROVED** [review]
LGTM. One thought, though, is that we should add a new element (versions? code? something else?) which can house the index version and others that we intend to add. Really transport version should be part of this too. This could even be a `VersionInformation` which is internally a Map of name to version number, and could be Nullable in the same way other node info is optional.

**Comment by thecoop** [comment]
Moving `transport_version` (and maybe `version`) is an incompatible change. How about we keep transport & index version separate, as 'core' node properties, and have a new `component_versions` field that has the pluggable version numbers in it as a subobject?

**Comment by rjernst** [comment]
That sounds good to me.

**Pull request merged**